synapse/storage/schema/main/delta/58/05cache_instance.sql.postgres

```diff
@@ -28,3 +28,8 @@ CREATE TABLE cache_invalidation_stream_by_instance (
 CREATE UNIQUE INDEX cache_invalidation_stream_by_instance_id ON cache_invalidation_stream_by_instance(stream_id);
 
 CREATE SEQUENCE cache_invalidation_stream_seq;
+
+-- The sequence needs to begin at 2 because a bunch of code assumes that
+-- get_next_id_txn will return values >= 2, cf this comment:
+-- https://github.com/matrix-org/synapse/blob/b93bd95e8ab64d27ae26841020f62ee61272a5f2/synapse/storage/util/id_generators.py#L344
+SELECT setval('cache_invalidation_stream_seq', 1);
```

Comment:
Based on the failure that can happen here thanks to `0`, should we update this example in the documentation to default to `1`

https://github.com/matrix-org/synapse/blob/fa2f3d8d0c915c6ebd84ef53526ac04c91fd4a19/docs/upgrade.md?plain=1#L425-L427

Reply:
Maybe doesn't matter because it's not a sequence number used as a stream ID with the `MultiWriterIdGenerator`

Reply:
Problem seems to be more related to code that call into `PostgresSequenceGenerator.get_next` than `get_next` itself.